cpp/src/arrow/compute/kernels/row_encoder.h

```diff
@@ -231,6 +231,26 @@ struct VarLengthKeyEncoder : KeyEncoder {
   std::shared_ptr<DataType> type_;
 };
 
+struct NullKeyEncoder : KeyEncoder {
+  void AddLength(const Datum&, int64_t batch_length, int32_t* lengths) override {
+    return;
+  }
```

Comment:
No need to `return` as it is a void method, empty braces will suffice.

Reply:
Similarly, for `AddLengthNull` and `EncodeNull`.

Reply:
Fixed